hs_core/serialization.py

```diff
@@ -0,0 +1,844 @@
+
+        # Get coverage (period)
+        for s, p, o in self._rmeta_graph.triples((None, None, rdflib.namespace.DCTERMS.period)):
+            coverage_lit = self._rmeta_graph.value(s, rdflib.namespace.RDF.value)
+            if coverage_lit is None:
+                msg = "Coverage value not found for {0}.".format(o)
+                raise GenericResourceMeta.ResourceMetaException(msg)
+            coverage = GenericResourceMeta.ResourceCoveragePeriod(str(coverage_lit))
+            self.coverages.append(coverage)
+
+        print("\t\tCoverages: ")
+        for c in self.coverages:
+            print("\t\t\t{0}".format(str(c)))
+
+        # Get relations
+        for s, p, o in self._rmeta_graph.triples((None, rdflib.namespace.DC.relation, None)):
+            for pred, obj in self._rmeta_graph.predicate_objects(o):
+                relation = GenericResourceMeta.ResourceRelation(obj, pred)
+                self.relations.append(relation)
+
+        print("\t\tRelations: ")
+        for r in self.relations:
+            print("\t\t\t{0}".format(str(r)))
+
+        # Get sources
+        for s, p, o in self._rmeta_graph.triples((None, rdflib.namespace.DC.source, None)):
+            for pred, obj in self._rmeta_graph.predicate_objects(o):
+                source = GenericResourceMeta.ResourceSource(obj, pred)
+                self.sources.append(source)
+
+        print("\t\tSources: ")
+        for r in self.sources:
+            print("\t\t\t{0}".format(str(r)))
+
+    def get_owner_uri(self):
+        """
+        Return the creator with the lowest order.
+
+        :return: String representing the URI of the owner of the resource.
+        """
+        min_order = sys.maxint
+        owner_uri = None
+        for c in self.creators:
+            if c.order < min_order:
+                min_order = c.order
+                owner_uri = c.uri
+                break
+        return owner_uri
+
+    def write_metadata_to_resource(self, resource):
+        """
+        Write metadata to resource
+
+        :param resource: HydroShare resource instance
+        """
+        if self.abstract:
+            resource.metadata.create_element('description', abstract=self.abstract)
+        if self.rights:
+            try:
+                resource.metadata.create_element('rights', statement=self.rights.statement)
+            except IntegrityError:
+                resource.metadata.update_element('rights', resource.metadata.rights.id,
+                                                 statement=self.rights.statement)
+        if self.language:
+            resource.metadata.update_element('language', resource.metadata.language.id,
+                                             code=self.language)
+        if self.creation_date:
+            res_created_date = resource.metadata.dates.all().filter(type='created')[0]
+            resource.metadata.update_element('date', res_created_date.id, start_date=self.creation_date)
```

Comment:
@hyi @ironfroggy @pkdash This update seems to take.  If I do a query:

    resource.metadata.dates.all().filter(type='created')[0]

Immediately after the update, I get back the updated value.  Yet the updated value is never seen in the web interface, even if I restart the Django and database containers.  This is the last thing I need to fix before moving on to implementing import of derived resource types, so your help is appreciated.  Thanks.